sdk/src/main/java/com/hedera/hashgraph/sdk/Client.java

```diff
@@ -106,6 +101,8 @@ public final class Client implements AutoCloseable {
     @Nullable
     private CompletableFuture<Void> networkUpdateFuture;
 
+    private Set<SubscriptionHandle> subscriptions = new HashSet<>();
```

Comment:
This is not thread safe. Also, there's not much reason it should be a set since SubscriptionHandle does not implement equals/hashCode and its addition/removal is handled internally. 
```suggestion
    private Collection<SubscriptionHandle> subscriptions = Collections.synchronizedList(new ArrayList<>());
```

Reply:
ArrayList would allow for duplicate records of the same instance to be added to the collection. The default Object.hashCode is all we need to avoid duplicates in a Set. I suggest that we settle for:

private Set<SubscriptionHandle> subscriptions = Collections.synchronizedSet(new HashSet<>());

Or is there any reason why duplicates should be allowed?

Reply:
I think it's fine to use set, just didn't think it was necessary since there shouldn't be duplicates because the object is always internally added/removed and not by the user. If we want to use a concurrent set, then I think we should use `ConcurrentHashMap.newKeySet()`. It's way more performant and has concurrent iterator support so we don't need to copy the collection to iterate. Only caveat is it's available since Java 8 and I'm not sure what the baseline Java version the SDK supports.